### PR TITLE
utilize @graphql-generate/cli for schema loading and generation

### LINKED ISSUE #117: use @graphql-codegen/cli instead of @graphql-codegen/core
as per [the documentation](https://github.com/dotansimha/graphql-code-generator/blob/6e045885d1ca66d69ce1ac069380b5c75cc5a91e/website/docs/getting-started/programmatic-usage.md) we can use cli instead and benefit from more feature parity with graphql-codegen. do you think it is a good idea?

### PR DESCRIPTION
**Pull request by salzhrani**
also solves some issues like url schemas and header config

Also added jest snapshots to track changes in the generated artifacts

fixes #117 #104 #85

### CODE REVIEW

**Comment by salzhrani** [comment]
I had to normalize string before hashing to get a consistent hash on windows or *nix.
another approach would that to remove the hash from the artifact comparing with the snapshot.

**Comment by piglovesyou** [comment]
Thank you, it's so fundamental, impacting on #104, #85 . Jest snapshot is a good idea. "normalize string before hashing" is worth it to make the code testable, unless it doesn't hit performance issue. I'm also interested in how this affects #117 .

@salzhrani So can you include a test for "accepting object in `config.schema`"?

File: src/lib/resolver-types.ts [hunk 1]
```diff
@@ -1,13 +1,9 @@
 import { extname } from 'path';
-import glob from 'globby';
-import slash from 'slash';
```
Comment (piglovesyou) [inline src/lib/resolver-types.ts:3]: Now we can remove "slash" from the dependencies.

**Comment by salzhrani** [comment]
I have benchmarked to find the effect of normalization and the result is minor
before: 11.1 microseconds
after: 14.2 microseconds

**Comment by piglovesyou** [comment]
Thank you for solving a lot of issues. Let me maintain the docs and types.ts for this later.

**Pull request merged**